cmake/compiler_settings.cmake

```diff
@@ -60,24 +60,31 @@ include(ccache)
 include(CheckCCompilerFlag)
 include(CheckCXXCompilerFlag)
 
-# Apply Spectre mitigations if available.
-set(SPECTRE_MITIGATION_FLAGS -mllvm -x86-speculative-load-hardening)
-check_c_compiler_flag("${SPECTRE_MITIGATION_FLAGS}"
-                      SPECTRE_MITIGATION_C_FLAGS_SUPPORTED)
-check_cxx_compiler_flag("${SPECTRE_MITIGATION_FLAGS}"
-                        SPECTRE_MITIGATION_CXX_FLAGS_SUPPORTED)
-if (SPECTRE_MITIGATION_C_FLAGS_SUPPORTED
-    AND SPECTRE_MITIGATION_CXX_FLAGS_SUPPORTED)
-  message(STATUS "Spectre 1 mitigations supported")
-  # We set this variable to indicate the flags are supported. It is
-  # empty otherwise.
-  set(OE_SPECTRE_MITIGATION_FLAGS ${SPECTRE_MITIGATION_FLAGS})
-  # TODO: We really should specify this only on the `oecore` target;
-  # however, the third-party mbed TLS build needs it set to, so we
-  # have to keep this here for now.
-  add_compile_options(${OE_SPECTRE_MITIGATION_FLAGS})
-else ()
-  message(WARNING "Spectre 1 mitigations NOT supported")
+# Apply Spectre mitigation if available.
+# The flag requires clang 8 or above.
+set(SPECTRE_MITIGATION_FLAGS -mspeculative-load-hardening)
```

Comment:
My preference is for the newer flags. @radhikaj Please confirm if we want to switch to the newer flag.
We must eventually switch to the newer flag; but clang-7 doesn't understand it. Is now a good time for the switch?

Reply:
Given that we're giving warning if users do not use clang-8, I think switching to the new flag along with the clang update is reasonable.  If users still want to use clang-7, the mitigation won't apply (`check_c_compiler_flag` will fail), which is fine as we no longer support clang-7.

Reply:
For this release, I would prefer that developers  building their application code with clang-7 should not get errors. So would prefer keeping flags in the exported targets same as they were for clang-7. @mingweishih  @anakrish is this possible?

Reply:
There is no error if developers use clang-7, but just the mitigation won't be applied.  This is due to that the build flags for application are inherent from the OE build. We can revisit this in the next release.

Reply:
Since clang-7 understands `-mllvm -x86...`, the mitigation will be applied as before.
We don't know if `-mllvm -x86..` and `-mspeculative-load-hardening` are equivalent.